src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,59 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+
+    static {
+        interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+    }
+
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    public <T> T getInstance(Class<T> interfaceClazz) {
+        Class<?> implementationClass = interfaceImplementations.get(interfaceClazz);
+        if (implementationClass == null) {
+            throw new RuntimeException("No implementation found for " + interfaceClazz.getName());
```

Comment:
When throwing an exception, it's recommended to add an informative message. The current message is informative, but ensure it provides enough context for debugging.